collections/_article/rendering-priorities-september-2024.md

```diff
@@ -0,0 +1,115 @@
+---
+title: "Godot Rendering Priorities: September 2024"
+excerpt: "4.3 is out. It's time for an update!"
+categories: ["progress-report"]
+author: Clay John
+image: /storage/blog/covers/rendering-priorities-september-2024.webp
+date: 2024-09-18 9:00:00
+---
+
+It's time again to update everyone on the activities of the rendering team. We have been hard at work fixing countless bugs and improving the renderer. Now that 4.3 is out, we have taken the chance to re-assess our priorities.
+
+## Get Involved
+
+As a reminder, we are always in need of more help. The rendering team is very small and none of us are working on the renderer full time. If you are interested in helping out please consider joining the #rendering channel on [chat.godotengine.org](chat.godotengine.org), joining the weekly meetings that place, or just opening a pull request!
+
+In addition to the priorities listed here, you can get started contributing by fixing bugs. We have been categorizing bugs by complexity and priority in the new [Rendering Issue triage project](https://github.com/orgs/godotengine/projects/78) to make it easier for new contributors to discover bugs that need attention.
+
+For a less organized overview on rendering priorities, you can also check out the [4.x Rendering Roadmap Github project](https://github.com/orgs/godotengine/projects/33). This project lists a lot of the long term tasks we would like to do that don't fit nicely in bug reports or in the blog post format. If you are performance-minded, there are also plenty of potential performance optimizations described in that project.
+
+## Update on Previous Priorities and 4.3
+
+The below are the priorities we identified in the last rendering priorities blog post. Some of these come from earlier blog posts as well. We are happy to report that almost all priorities from the previous posts are either finished, or ready to merge.
+
+* **Background pipeline compilation + Proper multi-threading in the RenderingDevice ([GH-90400](https://github.com/godotengine/godot/pull/90400)) (Ready to merge)**
+
+This work was finished during the 4.3 dev cycle, but due to the size and complexity we deferred merging it to 4.4. As a reminder, this PR implements the concept of an "ubershader" which is basically a version of the shader that contains all features and is compiled at load time. We pair the ubershader with the ability to compile specialized shader pipelines in the background. This way when we draw a new object, instead of freezing and waiting for the pipeline to compile, we fallback on the ubershader and compile the specialized shader in the background.
```

Comment:
```suggestion
This work was finished during the 4.3 dev cycle, but due to the size and complexity, we deferred merging it to 4.4. As a reminder, this PR implements the concept of an "ubershader" which is basically a version of the shader that contains all features and is compiled at load time. We pair the ubershader with the ability to compile specialized shader pipelines in the background. This way, when we draw a new object, instead of freezing and waiting for the pipeline to compile, we fallback on the ubershader and compile the specialized shader in the background.
```